### PR TITLE
Reselect

### PR DESCRIPTION
**Pull request by felire**
### Summary

I've implemented reselect library. I do not know how to try it. console.count command isn't working. I created the getEmptyInputSelector to avoid rerenders when the text change. But i repeat again, i don't know if this is working good. I would like that you see this and let me know if you seen something that you think it isn't okey.

There is no gifs to show in this PR

### CODE REVIEW

File: WBooks/src/app/screens/Search/index.js [hunk 1]
```diff
@@ -2,27 +2,25 @@ import React, { Component } from 'react';
...
 
 class Search extends Component {
-  filterBooks = (text, books) =>
-    books.filter(book => book.title.toLowerCase().includes(text.toLowerCase())).sort();
-
   render() {
-    const { text, books } = this.props;
-    const data = this.filterBooks(text, books);
+    const { books, isEmptyInput } = this.props;
+    console.log('write');
```
Comment (santinucera) [inline WBooks/src/app/screens/Search/index.js]: Remove this

File: WBooks/src/app/screens/Search/composition.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const highOrderComponent = (conditionalRenderingFn, EitherComponent) => Componen
   conditionalRenderingFn(props) ? <EitherComponent /> : <Component {...props} />;
 
 const isEmptyCondition = ({ data }) => !data.length || !data;
-const thereIsNoText = ({ text }) => !text;
+const thereIsNoText = ({ isEmptyInput }) => isEmptyInput;
```
Comment (santinucera) [inline WBooks/src/app/screens/Search/composition.js]: It's okey this?

File: WBooks/src/app/screens/Search/utils/selector.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import Immutable from 'seamless-immutable';
+
+const getBooks = state => state.books.books;
+const getText = state => state.search.text;
+
+export const getBooksSelector = createSelector(
+  getBooks,
+  getText,
+  (books, text) => {
+    return Immutable.asMutable(
```
Comment (santinucera) [inline WBooks/src/app/screens/Search/utils/selector.js]: Remove return

File: WBooks/src/app/screens/Search/composition.js (hunk 2 above)
Comment (felire) [reply WBooks/src/app/screens/Search/composition.js]: Yes, the function is receiving a boolean now, not a string

**Pull request merged**